website_sale_osc/controllers/main.py

```diff
@@ -0,0 +1,276 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+# Odoo, an open source suite of business apps
+# This module copyright (C) 2015 bloopark systems (<http://bloopark.de>).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import SUPERUSER_ID
+from openerp.addons.website_sale.controllers.main import website_sale
+from openerp.addons.web import http
+from openerp.addons.web.http import request
+from openerp.report import report_sxw
+
+
+class website_sale(website_sale):
+
+    """Add aditional functions to the website_sale controller."""
+
+    mandatory_billing_fields = ["name", "phone", "email", "street", "city",
+                                "country_id", "zip"]
+    optional_billing_fields = ["street2", "state_id", "vat", "vat_subjected"]
+
+    @http.route(['/shop/checkout'], type='http', auth='public',
+                website=True, multilang=True)
+    def checkout(self, **post):
+        """Checkout controller."""
+        # if onestepcheckout is deactivated use the normal checkout
+        if not request.website.use_osc:
+            return super(website_sale, self).checkout()
+
```

Comment:
These extra empty lines in between a method makes the code less readable (although this can be subjective), but Python style guides recommend not to put them.